plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,278 @@
+                )
+                alerts_has_more_pages = True
+
+            alerts, state = self._dedupe_and_get_last_time(alerts, state, start_alert_time, observations=False)
+        else:
+            self.logger.info("No alerts retrieved for time period searched...")
+            state[LAST_ALERT_TIME] = end_alert_time
+        self.logger.info(f"{LAST_ALERT_TIME} set to: {state[LAST_ALERT_TIME]}, has_more_pages={alerts_has_more_pages}")
+
+        return alerts, alerts_has_more_pages, state
+
+    def trigger_observation_search_job(self, start_time: str, end_time: str, state: Dict[str, str]) -> Tuple[str, Dict]:
+        endpoint = f"api/investigate/v2/orgs/{self.connection.org_key}/observations/search_jobs"
+        search_params = {
+            "rows": PAGE_SIZE,
+            "start": 0,
+            "fields": ["*"],
+            "criteria": {"observation_type": OBSERVATION_TYPES},
+            "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+            "time_range": {"start": start_time, "end": end_time},
+        }
+        url = f"{self.connection.base_url}/{endpoint}"
+        self.logger.info(f"Triggering observation search using parameters {search_params['time_range']}")
+        observation_job_id = self.connection.request_api(url, search_params).get("job_id")
+
+        if observation_job_id:
+            self.logger.info(
+                f"Saving observation job ID {observation_job_id} to the state. "
+                f"Will query this after polling for alerts..."
+            )
+            state[LAST_OBSERVATION_JOB] = observation_job_id
+
+            if not state.get(LAST_OBSERVATION_TIME):
+                # First run of trying to get observations, save the start time for usage in `_dedupe_and_get_last_time`
+                self.logger.info(f"No {LAST_OBSERVATION_TIME} in the state, saving checkpoint as {start_time}")
+                state[LAST_OBSERVATION_TIME] = start_time
+
+        return observation_job_id, state
+
+    def get_observations(self, job_id: str, state: Dict[str, str]) -> Tuple[list, bool, Dict[str, str]]:
+        observations, has_more_pages = [], False
+        endpoint = f"api/investigate/v2/orgs/{self.connection.org_key}/observations/search_jobs/{job_id}/results"
+
+        # Strange CB API behaviour, unless rows param is specified it only returns 10 results
+        url = f"{self.connection.base_url}/{endpoint}?rows={PAGE_SIZE}"
+        self.logger.info(f"Get observation results from saved ID: {job_id}")
+        observation_json = self.connection.request_api(
+            url,
+            request_method="GET",
+        )
+        if observation_json.get("contacted") != observation_json.get("completed"):
+            self.logger.info("Job is not yet finished running, will get results in next task execution...")
+            has_more_pages = True  # trigger again as it should be finished imminently (jobs run for a max of 3 minutes)
```

Comment:
Had a read at the docs and they do state they can return an incomplete difference of 1 if the job is during a high activity period. I've included a 4 minute grace period to pull available results. 
> The job will be complete when contacted == completed in the response however during high usage a searcher may fail leaving a difference of 1. To prevent an infinite loop, ensure you add a timeout of 3 mins as a job's maximum active time is limited to 3 mins.